### PR TITLE
added extraConfigmapMounts support to agent

### PR DESCRIPTION
**Pull request by arpitjindal97**
It will give the user flexibility to use Mutual TLS feature
Collector is configured properly to adapt it but agent wasn't

### CODE REVIEW

File: charts/jaeger/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 1.16.0
 description: A Jaeger Helm chart for Kubernetes
 name: jaeger
-version: 0.18.3
+version: 0.18.4
```
Comment (cpanato) [inline charts/jaeger/Chart.yaml]:
```suggestion
version: 0.19.0
```

Comment (arpitjindal97) [reply charts/jaeger/Chart.yaml]: For now, Let it be `0.18.4`, I have added **Ingester** support also to this chart. I will create another PR for that. We can change the version to `0.19.0` in that.

I'm currently trying out the Ingester part. I will make that PR next week hopefully :)

File: charts/jaeger/values.yaml [hunk 2]
```diff
@@ -121,6 +121,12 @@ agent:
   ## Additional pod labels
   ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
   podLabels: {}
+  extraConfigmapMounts: []
+  # - name: jaeger-tls
+  #   mountPath: /tls
+  #   subPath: ""
+  #   configMap: jaeger-tls
+  #   readOnly: true
```
Comment (naseemkullah) [inline charts/jaeger/values.yaml]: Perhaps we should remove this example? It is not best practice to store TLS key as configmap, rather it should be stored as a secret.

Comment (arpitjindal97) [reply charts/jaeger/values.yaml]: I have made a commit resolving this issue and introducing few new things

**Comment by naseemkullah** [comment]
Can you remove please remove ingester related changes from here? 1 logical change per PR is preferable for my tired eyes. :)

Please keep the scope of this PR to adding `extraConfigmapMounts` and perhaps stretch to adding `extraSecrets` if needed.

**Comment by naseemkullah** [comment]
As well as the `cassandra.keyspace` changes, please open a separate PR and describe the purpose of the change as it is not immediately obvious what the goal is. Smaller scoped PRs really help with reviewing 😬

File: charts/jaeger/templates/agent-ds.yaml [hunk 3]
```diff
@@ -84,6 +84,30 @@ spec:
...
         resources:
 {{ toYaml .Values.agent.resources | indent 10 }}
+        volumeMounts:
+        {{- range .Values.agent.extraConfigmapMounts }}
+          - name: {{ .name }}
+            mountPath: {{ .mountPath }}
+            subPath: {{ .subPath }}
+            readOnly: {{ .readOnly }}
+        {{- end }}
+        {{- range .Values.agent.secretMounts }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/agent-ds.yaml]: nit: Can we call these `extraSecretMounts` please? Just to match naming of `extraConfigmapMounts`

File: charts/jaeger/templates/cassandra-schema-job.yaml [hunk 4]
```diff
@@ -46,10 +46,8 @@ spec:
             secretKeyRef:
               name: {{ if .Values.storage.cassandra.existingSecret }}{{ .Values.storage.cassandra.existingSecret }}{{- else }}{{ include "jaeger.fullname" . }}-cassandra{{- end }}
               key: password
-      {{- if .Values.schema.keyspace }}
         - name: KEYSPACE
-          value: {{ .Values.schema.keyspace | quote }}
-      {{- end }}
+          value: {{ .Values.storage.cassandra.keyspace | quote }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/cassandra-schema-job.yaml]: Please include in a seperate PR describing the problem this solves as it is not immediately obvious to me! Thanks

File: charts/jaeger/templates/ingester-deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,149 @@
+{{- if .Values.ingester.enabled -}}
```
Comment (naseemkullah) [inline charts/jaeger/templates/ingester-deploy.yaml]: Definitely something we want to do! But Please open a separate PR where we tackle just this.

File: charts/jaeger/templates/agent-ds.yaml (hunk 3 above)
Comment (arpitjindal97) [reply charts/jaeger/templates/agent-ds.yaml]: I'm closing this PR. I will create PRs one by one describing purpose of each

File: charts/jaeger/templates/cassandra-schema-job.yaml (hunk 4 above)
Comment (arpitjindal97) [reply charts/jaeger/templates/cassandra-schema-job.yaml]: Here is the discussion https://github.com/jaegertracing/jaeger-operator/pull/837

**Comment by naseemkullah** [comment]
Thanks @arpitjindal97

**Pull request closed without merging**